v2/cf-nodejs-logging-support/src/lib/logger/source-utils.ts

```diff
@@ -22,23 +27,58 @@ export class SourceUtils {
         return SourceUtils.instance;
     }
 
-    getFieldValue(source: Source, record: any): string | undefined {
+    getFieldValue(fieldName: string, source: Source, record: any, now: Date): string | undefined {
         switch (source.type) {
             case "static":
                 return source.value;
             case "env":
                 if (source.path) {
-                    return NestedVarResolver.resolveNestedVariable(process.env, source.path);
+                    // clone path to avoid deleting path in resolveNestedVariable()
+                    const clonedPath = [...source.path];
+                    return NestedVarResolver.resolveNestedVariable(process.env, clonedPath);
                 }
                 return process.env[source.name!];
             case "config-field":
+                if (fieldName == "correlation_id" && !uuidCheck.exec(record[source.name!])) {
+                    return;
+                }
                 return record[source.name!];
+            case "meta":
+                if (now == null) {
+                    return;
+                }
+                if (fieldName == "request_received_at") {
```

Comment:
Avoid hard-coding field names. This has to be done in config. For `meta` fields it is ok to use a separate meta `name` field in the source config to identify the meta field. However this should be decoupled from the `fieldName`. See also my previous comment.